### DISCUSSION TITLE
Moving tab or a tab group by drag-and-drop

### DISCUSSION THREAD

**Discussion by fuji2021** [mbnuqw/sidebery#1563] [General]
I have a big tab group, inside which are smaller tab groups (a multi-level tree of tabs). I'm trying to move the inner tab groups to outside the parent tab group. Currently it seems the only way to do this is to collapse the inner group and drag it all the way down, or to create a new panel. While dragging the group, other tabs suddenly get active (where the mouse hovers for a bit longer) and the dragging process has to be redone again. 

This seems like an ineffective way to move an inner tab group to outside the parent tab group. Is this a bug, or am I requesting a new feature? Not sure.

**Comment by mbnuqw** [comment]
To move descendant tabs from the group, you need to select them all (using the ctrl/shift+click or right-click selection if you don't use native context menu) and then move them with drag and drop. Dragging the parent tab will select the entire branch.

**Comment by fuji2021** [comment]
@mbnuqw Thanks! Does it mean I have to keep dragging through the parent group's tabs (which may be a few dozens or hundreds) in order to move out an inner tab group? That's what I tried and gave up. I made a new panel instead. The reason is that while dragging a group of tabs, other tabs that are on the way are randomly activated. The worflow seems feasible for only groups of a handful of tabs.

**Comment by albino1** [comment]
There's already a way to effectively copy by selecting a bunch of tabs and Duplicating, maybe there could be a way to do a Cut style command where instead they get removed and then you paste them into a different location?

**Comment by fuji2021** [comment]
The whole "drag to move" thing seems to be broken. The addon is great but I wish there was a good way to perform these common tasks. 

I just tried moving a single tab to a different location. If I pause a while, the tab closest to the cursor is suddenly activated. If I have a lot of tabs, I need to pause every now and then to find the right position to drag-and-drop a tab and this means a lot of unexpected things are happening in the browser and I lose the original tab being dragged. Can't it be configured such that a tab is activated only when actively clicked and not in the middle of dragging other tabs?

This is made worse by the fact that every time I want to open a new tab (by right clicking on a tab and choosing "New Tab"), the new tab is created at the end of the tab list. I want the new tab to be right next to the tab where I am working. Then, to bring the new tab to where I intended it to be, I need to do a lot of dragging, pausing, finding the tab, and dragging again.

Opening a new tab and moving some tabs around should be among the most frequent actions in a web browser, and I hope this issue could be handled better.

**Comment by albino1** [comment]
Under `Settings -> Drag and Drop` disable `Activate tab on hover`:
<details>
  <summary>Tab Hover Settings</summary>

![side1](https://user-images.githubusercontent.com/12164957/216491330-b5389df9-d4c4-4781-aa00-be5bc2ab408f.jpg)

</details>

Then under `Settings -> Tabs -> Position of new tab` change those to whatever your preferred locations are.  For example, here are mine:

<details>
  <summary>New Tab Settings</summary>

![side2](https://user-images.githubusercontent.com/12164957/216491540-526ae9ca-1819-41b0-a12b-afd582f21ffc.jpg)

</details>

**Comment by mbnuqw** [comment]
@fuji2021, sorry, I didn't read your initial post carefully. As @albino1 mentioned, you can configure activation of tabs on drag and drop - increase delay or completely turn it off.

Also, you can expand/fold branches while dragging (check settings in the same section). Example:

https://user-images.githubusercontent.com/6276694/216617838-73142d4b-c2f0-405e-8b35-b5996428e209.mp4

**Comment by fuji2021** [comment]
Thanks @albino1 and @mbnuqw ! My settings options look different than yours but I found the relevant options and will try around.

Meanwhile, I think it'd be nice to have an easy action for "move-out-of-parent-tree." Thanks for making a great tool!

**Comment by albino1** [comment]
If your settings look different then you're probably on the old stable v4 version of Sidebery from the Firefox store which isn't really receiving updates any longer.  Most of the people on here are using the beta branch of v5 which is vastly improved, getting better all the time, and has been rock solid stability-wise.  You're welcome to try it yourself by installing the XPI directly from here listed under Assets:

https://github.com/mbnuqw/sidebery/releases/tag/v5.0.0b31
